### ISSUE TITLE
False positive with `ParameterPackage`

### ISSUE THREAD

**Issue by Stephan202** [google/error-prone#1177]
Consider this following code:
```java
import java.util.concurrent.Callable;
import java.util.concurrent.RunnableScheduledFuture;
import java.util.concurrent.ScheduledThreadPoolExecutor;

class Executor extends ScheduledThreadPoolExecutor {
    Executor() {
        super(1);
    }

    @Override
    protected <V> RunnableScheduledFuture<V> decorateTask(
            Runnable runnable, RunnableScheduledFuture<V> task) {
        return task;
    }

    @Override
    protected <V> RunnableScheduledFuture<V> decorateTask(
            Callable<V> callable, RunnableScheduledFuture<V> task) {
        return task;
    }
}
```

Compile this code as follows:
```sh
javac \
  -XDcompilePolicy=simple \
  -processorpath error_prone_core-2.3.2-with-dependencies.jar \
  '-Xplugin:ErrorProne -XepDisableAllChecks -Xep:ParameterPackage:WARN' \
  Executor.java
```

The following false positive results:
```
Executor.java:11: warning: [ParameterPackage] Method parameter has wrong package
    protected <V> RunnableScheduledFuture<V> decorateTask(
                                             ^
    (see https://errorprone.info/bugpattern/ParameterPackage)
  Did you mean 'Runnable runnable, java.util.concurrent.RunnableScheduledFuture task) {'?
1 warning
```

Interestingly the code complains only about the `Runnable` overload; not the `Callable` overload.
